### PR TITLE
Information override

### PR DESCRIPTION
**Pull request by talbotmcinnis**
My first Pull Request repo to repo.  Not sure if the best practice is to PR on by branch, or for me to PR to my own master and then PR my master to your master?  Let me know if this is not the best practice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,6 +186,8 @@ WEATHER Mountain Range 251.000, VOICE en-US-Standard-E
 - [**win-media**](./crates/win-media) - Bindings to a subset of the Windows Runtime (separate crate to reduce the compile time during development).
 - [**win-tts**](./crates/win-tts) - A library to convert text to speech using the Window Runtime.
 
+- 
+
```
Comment (rkusa) [inline README.md:190]: Looks like an untended change that could potentially be removed?

**Review by rkusa — COMMENTED** [review]
Thanks a lot for the PR! As far as I am aware, both your own branch `talbotmcinnis:InformationOverride` or your `talbotmcinnis:master` are fine as a source for the PR. I personally use both methods depending on whether I am potentially working on multiple PRs for the same repo or not.
Changes look good 👍, but I am afraid that recent changes in this repo resulted a merge conflict now. Would you mind rebasing your branch onto the recent `master`? Let me know if need help with that.

**Comment by talbotmcinnis** [comment]
Rebase and conflicts resolved.  Unintended readme change reverted as well.

**Comment by rkusa** [comment]
Thank you very much for updating the PR and the contribution 👍
I've added the `hacktoberfest` topic to repo in case you want to participate in https://hacktoberfest.digitalocean.com

**Comment by talbotmcinnis** [comment]
Ok that’s great!  Thanks for all the feedback along the way.  

I assume the next best step will be to try to extract/test the order-independent option regex.  If it works already, great.  If not I had a solution written up that I should be able to squeeze in now.

I’ve got a few other projects in the air so may be a dew days before I can dig in.  Hacktoberfest looks great, not sure I’ll have the time but I will try to keep an eye on it!

**Comment by rkusa** [comment]
> I assume the next best step will be to try to extract/test the order-independent option regex. If it works already, great. If not I had a solution written up that I should be able to squeeze in now. I’ve got a few other projects in the air so may be a dew days before I can dig in.

Yes that would be a good next step, but I think regex isn't the best solution here anymore. I think using a parser combinator like https://github.com/Geal/nom might be a good option instead. But don't feel obliged to look into that.

> Hacktoberfest looks great, not sure I’ll have the time but I will try to keep an eye on it!

I just saw that this was about four PRs, I always thought one is enough. So just this PR is sadly not enough to claim a tree or tshirt.

**Comment by talbotmcinnis** [comment]
I spent just a few minutes looking at nom.  I admit I wasn't able to see
the benefit versus a slight restructured and simplified regex approach.
Most examples of nom that I encountered seemed to have syntax as cryptic as
regex, but less universally known.

To sketch out what I'm comparing it to, I have committed a regex based
approach in this commit:  This only tackles one of the extract functions,
as it is meant just as an example for now.

https://github.com/talbotmcinnis/DATIS/commit/5b1cdc6b386052f7d245f95f6b7eba8d13c26a3f

Still 100% your project so I'm offering this for consideration but can
still look further into nom or something like it.

Cheers,

**Comment by rkusa** [comment]
@talbotmcinnis You are probably right that the same can be achieved with regex and is more readable in the end. The main reason `nom` (or any other parser generator approach) sounds intriguing to me is that we get some useful error messages for free. The current error behavior is simply that the station does not work (or settings are ignored), it would be nice to have some more information about why and about possible mistakes in this settings string.

But also this can be achieved with regex. I am thinking about something like `(,([A-Z]+) [^,])*$` to get all key value pairs. We could then have a `match` that goes through all known keys, parses their values, but also have a fallback to notify about unknown keys:

```rust
for (key, value) in pairs {
  match (key) {
    "TRAFFIC" => /* parse traffic value, and log a useful error if the value does not match the expected pattern */,
    "INFO" => /* parse info value, and log a useful error if the value does not match the expected pattern */,
    _ => /* log warning about unknown key */,
  }
}
```

What do you think about that approach?

**Comment by talbotmcinnis** [comment]
Love this direction, let me give it a try!

**Comment by rkusa** [comment]
Just a random thought, since the next release is going to be a breaking one anyway, we could also think about changing the syntax a bit, e.g. by including a `=` between keys and values. Example:

```diff
- ATIS Kutaisi 251.000, TRAFFIC 252.000, VOICE WIN, INFO Q
+ ATIS Kutaisi 251.000, TRAFFIC=252.000, VOICE=WIN, INFO=Q
```

Or even:

```diff
- ATIS Kutaisi 251.000, TRAFFIC 252.000, VOICE WIN, INFO Q
+ ATIS Kutaisi 251.000 TRAFFIC=252.000 VOICE=WIN INFO=Q
```

Not sure if one of those variants is easier to use for non-technical users of the mod. WDYT?

**Comment by rkusa** [comment]
I was wrong, the next version is not a major release, so not worth worrying about my previous message yet.

**Comment by talbotmcinnis** [comment]
I'll comment anyway in advance of the change.  I think the best solution
would be the most human readable.  I remain optimistic we can get the
extractor to function without requiring specific syntax like equal signs.
If we expose some reason to require a more extractor-friendly syntax, then
we could do so, but as is I think it was a clever idea to allow the human
readable format to double as the info for the program.  In this version I
will certainly support it.  If I can easily allow the = separator, I'll
toss that in but I will toss my vote to continuing with space + comma
separation.

**Pull request merged**